### ISSUE TITLE
Add default foliage/water color when world is not loaded

### ISSUE THREAD

**Issue by Naezr** [MehVahdJukaar/polytone#102]
I use a resourcepack that, with Polytone, makes items in inventory use the colour of the current biome. 

In the game itself this works perfectly, but in the flat world customisation menu (and in any other cases where the world is not loaded) items are coloured white, as there is no biome. 

Suggestion: Assign a default biome for the main menu, such as a plains, so that items are coloured in its colours and not white.

Screenshot:

![изображение](https://github.com/user-attachments/assets/2a248e75-295c-43c0-a66a-1d262512ee8d)

Resourcepacks I'm using: https://modrinth.com/user/mineland

**Comment by MehVahdJukaar** [comment]
You are correct, you say "as if there is no biome", that's precisely because there isn't. Biomes ara data driven and are loaded with a world as they do depend on the world data, as all datapacka do. This means that there exactly are NO biome objects before a world is loaded

**Comment by MehVahdJukaar** [comment]
All colormaps do support a default color thing is I cannot load their definition yet as their internals depend on biomes

**Comment by Naezr** [comment]
> This means that there exactly are NO biome objects before a world is loaded

That sounds like a lead. I don't know how this mod works, and I'm probably going to say some nonsense, but still. Is it possible to add a check to the function that gets the biome color to see if it's missing? As far as I guess, the mod tries to color these white textures with the biome color, and if it fails, no coloring occurs. That is, the biome color variable is zero or something like that. Is it possible to add a check for this and implement some kind of default color instead of zero? 

Once again, I don't know much about Minecraft modding or Java, so don't take this too seriously.

**Comment by MR-CHOO** [comment]
It’s most likely bug with the pack that you are using, when no biome is loaded it will load a color from the colormap, I’m not sure from which value though, might be 256 at the end. Anyways whichever value it’s loading it probably just white on the colormap in the pack. I’ll have to test this with my pack later though to confirm.

**Comment by mineland-2048** [comment]
> Anyways whichever value it’s loading it probably just white on the colormap in the pack.

Creator of the pack theyre using there. I noticed the superflat world screen being weird aswell during testing, but since i used the default `minecraft:grass_color` provider for the colormap im not able to set a default value for it. It should probably be set somewhere inside the mod manually though.

snippet thats causing the problem (broken in 1.21.4 for some reason, but still works on older versions)
```json
{
   "targets": [ "..." ],
   "colormap": {
      "0": "minecraft:grass_color"
   }
}
```

since its using a builting and not one thats defined somewhere else in the pack it has no `default_color` I assume


> All colormaps do support a default color thing is I cannot load their definition yet as their internals depend on biomes

im not sure if this is a big roadblock, but id suggest for the internal ones to just make a default constant value on the spot based on the most common colors of these things in the game (grass/foliage using the grass block / leaves items color, water using some biome's color already like the plains', etc)

From the 1.21.4 item models:
* Grass: #7cbd6b (from downfall 0.5, temperature 1.0)
* Foliage: int -12012264
* Water: #3F76E4 (from most biomes)

The other built in ones will not have the same luck for this approach to be plausible though, but idk
